docs/content/TextInput.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: TextInput
+status: Alpha
+---
+
+import {PropsTable} from '../src/props-table'
+
+TextInput is a form component to add default styling to the native text input.
+
+**Note:** Don't forget to set `aria-label` to make the TextInput accessible to screen reader users.
+
+## Default example
+
+```jsx live
+<>
+  <TextInput aria-label="Zipcode" name="zipcode" placeholder="Zipcode" autoComplete="postal-code" />
+
+  <TextInput sx={{ml: 4}} aria-label="City" name="city" placeholder="City" contrast />
+
+  <TextInput
+    sx={{ml: 4}}
+    icon={SearchIcon}
+    aria-label="Zipcode"
+    name="zipcode"
+    placeholder="Find user"
+    autoComplete="postal-code"
+  />
+</>
+```
+
+## Component props
+
+Native `<input>` attributes are forwarded to the underlying React `input` component and are not listed below.
+
+<PropsTable>
+  <PropsTable.Row name="aria-label" type="string" description="Required. Allows input to be accessible." />
+  <PropsTable.Row name="block" type="boolean" description="Adds `display: block` to element" />
+  <PropsTable.Row name="contrast" type="boolean" description="Changes background color to a higher contrast color" />
+  <PropsTable.Row
+    name="variant"
+    type="string"
+    description="Can be either `small` or `large`. Creates a smaller or larger input than the default."
+  />
+  <PropsTable.Row name="width" type="string | number" description="Set the width of the input" />
```

Comment:
I wonder if we should accept an array for enum types. Example:

```
type={['string', 'number']}
```

Requiring people to manually type `|` could introduce some inconsistency. For example people might write, `string or number`, or `string, number`, instead of `string | number`

Reply:
I don't feel too strongly about this. Just an idea. Happy to ship what you have here and see if we end up with inconsistencies

Reply:
I like the mindset of reducing inconsistency (ultimately I want us to be able to do that through generating the docs). What would you expect to happen when one of the types needs to be linked?

Reply:
I was thinking elements in the array could be strings of JSX elements:
```
type={['string', <Link href='#>number</Link>]}
```

Reply:
Gotcha, thanks. I think I'm going to hold off on that for now — in part because `|` isn't even the only type operator that could apply there and I want to think about that, and in part because this still feels like forward progress.